src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/VerifyThingAttachedToCertificate.java

```diff
@@ -89,7 +89,7 @@ public Boolean apply(VerifyThingAttachedToCertificateDTO dto) {
 
             return verifyLocally(thing, dto.getCertificateId());
         } catch (CloudServiceInteractionException e) {
-            return false;
+            return verifyLocally(thing, dto.getCertificateId());
```

Comment:
This is better, but we still need to ensure we go to the cloud if we don't have anything in the local registry, regardless of network state.

Refer to my comment in the NetworkState class. We cannot rely on it to be accurate. Right now we have no way to authenticate a client if the `isNetworkUp` never returns true